src/pages/index.js

```diff
@@ -1,211 +1,269 @@
+      api.getInitialCards(),
+      api.getUserInfo(),
+    ]);
+
+    cardSection.setItems(cards);
+    user.setUserInfo({
+      name: userInfo.name,
+      about: userInfo.about,
+    });
+    user.setUserAvatar({
+      avatar: userInfo.avatar,
+    });
+  } catch (err) {
+    console.error("Initialization error:", err);
+  }
+};
+
+initialize();
 
-// ! EVENT LISTENERS FOR POPUPS
+// ! EVENT HANDLERS
+
+// Handle profile edit form submission
+function handleProfileEditSubmit(profileData) {
+  return handleSubmit(() => {
+    return api
+      .updateProfileInfo({
+        name: profileData.title,
+        description: profileData.subtitle,
+      })
+      .then((updatedData) => {
+        user.setUserInfo(updatedData);
+      });
+  }, profileEditPopup);
+}
 
+// Add card handler
+function handleAddCardSubmit(inputValues) {
+  return handleSubmit(() => {
+    return api
+      .addCard({ name: inputValues.title, link: inputValues.image })
+      .then((newCardData) => {
+        const newCardElement = createCard(
+          newCardData,
+          handleCardImageClick,
+          handleDeleteClick,
+          handleLikeClick
+        );
+        cardSection.addItem(newCardElement);
+      });
+  }, addCardPopup);
+}
+
+// Handle profile image form submission
+function handleProfileImageSubmit(newImageData) {
+  return handleSubmit(() => {
+    return api
+      .updateProfileAvatar({ avatar: newImageData.image })
+      .then((res) => {
+        user.setUserAvatar({ avatar: res.avatar });
+      });
+  }, profileImagePopup);
+}
+
+// Handle card delete confirmation
+function handleDeleteClick(card) {
+  if (!card._cardElement) {
+    console.error("Card element is null for card:", card);
+  }
+
+  cardToDelete = card;
+  deleteCardPopup.open();
+}
+
+// Handle card like button
+function handleLikeClick(card) {
+  const newLikeState = !card.isLiked;
+  card.updateLikeState(newLikeState);
+  card.isLiked = newLikeState;
+
+  if (newLikeState) {
+    api.likeCard(card.getId()).catch((err) => console.error(err));
+  } else {
+    api.unlikeCard(card.getId()).catch((err) => console.error(err));
+  }
+}
+
+// Handle card delete confirmation submit
+function handleConfirmDelete() {
+  return handleSubmit(
+    () => {
+      if (!cardToDelete) {
+        console.error("No card to delete.");
+        return Promise.resolve();
+      }
+
+      return api
+        .deleteCard(cardToDelete.getId())
+        .then(() => {
+          // Verify if cardToDelete still has a valid element
+          if (cardToDelete._cardElement) {
+            cardToDelete.removeCard();
+          } else {
+            console.warn("Card element is null");
+          }
+          cardToDelete = null;
+          return Promise.resolve();
+        })
+        .catch((error) => {
+          console.error("Error deleting card:", error);
+          return Promise.reject(error);
+        })
+        .finally(() => {
+          deleteCardPopup.close();
```

Comment:
` deleteCardPopup.close();`


Please, delete it

You have everything in `handleSubmit` in a `then` block

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well